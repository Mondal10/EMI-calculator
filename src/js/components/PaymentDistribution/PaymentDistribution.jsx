import React from 'react';

import styled from 'styled-components';

const Container = styled.div`
  font-size: 20px;
  padding: 2%;
  border-radius: 20px;
  color: #ededf2;
  margin-top: 10px;
`;

const Table = styled.table`
  border-collapse: collapse;
  width: 100%;
  text-align: center;
  margin: 10px auto;
`;

const TableHead = styled.th`
  padding-top: 12px;
  padding-bottom: 12px;
  text-align: center;
  border: 1px solid #7c7e80;
`;

const TableData = styled.td`
  border: 1px solid #7c7e80;
  padding: 8px;
`;

function PaymentDistribution({ monthlyDistribution, yearlyDistribution }) {
  // Convert to INR format with symbol
  const rupeeFormatter = new Intl.NumberFormat('en-IN', {
    style: 'currency',
    currency: 'INR',
    minimumFractionDigits: 0
  })

  return (
    <Container>
      <Table>
        <thead>
          <tr>
            <TableHead>Years</TableHead>
            <TableHead>Principal</TableHead>
            <TableHead>Interest</TableHead>
            <TableHead>EMI</TableHead>
            <TableHead>Balance</TableHead>
          </tr>
        </thead>
        <tbody>
          {
            yearlyDistribution.map(({ yearlyInterest, yearlyPrinciple, balance }, i) => (
              <tr key={`Year: ${i + 1}`}>
                <TableData>Year: {i + 1}</TableData>
                <TableData>{rupeeFormatter.format(yearlyPrinciple)}</TableData>
                <TableData>{rupeeFormatter.format(yearlyInterest)}</TableData>
                <TableData>{rupeeFormatter.format(yearlyPrinciple + yearlyInterest)}</TableData>
                <TableData>{rupeeFormatter.format(balance)}</TableData>
              </tr>
            ))
          }
        </tbody>
      </Table>
      <Table>
        <thead>
          <tr>
            <TableHead>Months</TableHead>
            <TableHead>Principal</TableHead>
            <TableHead>Interest</TableHead>
            <TableHead>EMI</TableHead>
            <TableHead>Balance</TableHead>
          </tr>
        </thead>
        <tbody>
          {
            monthlyDistribution.map(({ monthlyInterest, monthlyPrinciple, balance }, i) => (
              <tr key={`Month: ${i + 1}`}>
                <TableData>Month: {i + 1}</TableData>
                <TableData>{rupeeFormatter.format(monthlyPrinciple)}</TableData>
                <TableData>{rupeeFormatter.format(monthlyInterest)}</TableData>
                <TableData>{rupeeFormatter.format(monthlyPrinciple + monthlyInterest)}</TableData>
                <TableData>{rupeeFormatter.format(balance)}</TableData>
              </tr>
            ))
          }
        </tbody>
      </Table>
    </Container>
  )
}

export default PaymentDistribution;
