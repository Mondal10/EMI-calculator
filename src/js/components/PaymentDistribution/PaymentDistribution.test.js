import React from 'react';
import renderer from 'react-test-renderer';
import PaymentDistribution from './PaymentDistribution';

describe("Payment Distribution Snapshot", () => {
  const monthlyDistribution = [
    { "monthlyInterest": 8750, "monthlyPrinciple": 58500, "balance": 941500 },
    { "monthlyInterest": 8238, "monthlyPrinciple": 59012, "balance": 882488 },
    { "monthlyInterest": 7722, "monthlyPrinciple": 59528, "balance": 822960 },
    { "monthlyInterest": 7201, "monthlyPrinciple": 60049, "balance": 762911 },
    { "monthlyInterest": 6675, "monthlyPrinciple": 60575, "balance": 702336 },
    { "monthlyInterest": 6145, "monthlyPrinciple": 61105, "balance": 641231 },
    { "monthlyInterest": 5611, "monthlyPrinciple": 61639, "balance": 579592 },
    { "monthlyInterest": 5071, "monthlyPrinciple": 62179, "balance": 517413 },
    { "monthlyInterest": 4527, "monthlyPrinciple": 62723, "balance": 454690 },
    { "monthlyInterest": 3979, "monthlyPrinciple": 63271, "balance": 391419 },
    { "monthlyInterest": 3425, "monthlyPrinciple": 63825, "balance": 327594 },
    { "monthlyInterest": 2866, "monthlyPrinciple": 64384, "balance": 263210 },
    { "monthlyInterest": 2303, "monthlyPrinciple": 64947, "balance": 198263 },
    { "monthlyInterest": 1735, "monthlyPrinciple": 65515, "balance": 132748 },
    { "monthlyInterest": 1162, "monthlyPrinciple": 66088, "balance": 66660 },
    { "monthlyInterest": 583, "monthlyPrinciple": 66667, "balance": 0 }
  ];

  const yearlyDistribution = [
    { "yearlyInterest": 70210, "yearlyPrinciple": 736790, "balance": 263210 },
    { "yearlyInterest": 5783, "yearlyPrinciple": 263217, "balance": 0 }
  ];

  it('Payment Distribution rendered successfully', () => {
    const tree = renderer.create(<PaymentDistribution monthlyDistribution={monthlyDistribution} yearlyDistribution={yearlyDistribution} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
