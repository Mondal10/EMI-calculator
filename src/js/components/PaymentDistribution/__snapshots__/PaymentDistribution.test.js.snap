// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Payment Distribution Snapshot Payment Distribution rendered successfully 1`] = `
.c0 {
  font-size: 20px;
  padding: 2%;
  border-radius: 20px;
  color: #ededf2;
  margin-top: 10px;
}

.c1 {
  border-collapse: collapse;
  width: 100%;
  text-align: center;
  margin: 10px auto;
}

.c2 {
  padding-top: 12px;
  padding-bottom: 12px;
  text-align: center;
  border: 1px solid #7c7e80;
}

.c3 {
  border: 1px solid #7c7e80;
  padding: 8px;
}

<div
  className="c0"
>
  <table
    className="c1"
  >
    <thead>
      <tr>
        <th
          className="c2"
        >
          Years
        </th>
        <th
          className="c2"
        >
          Principal
        </th>
        <th
          className="c2"
        >
          Interest
        </th>
        <th
          className="c2"
        >
          EMI
        </th>
        <th
          className="c2"
        >
          Balance
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          className="c3"
        >
          Year: 
          1
        </td>
        <td
          className="c3"
        >
          ₹ 7,36,790
        </td>
        <td
          className="c3"
        >
          ₹ 70,210
        </td>
        <td
          className="c3"
        >
          ₹ 8,07,000
        </td>
        <td
          className="c3"
        >
          ₹ 2,63,210
        </td>
      </tr>
      <tr>
        <td
          className="c3"
        >
          Year: 
          2
        </td>
        <td
          className="c3"
        >
          ₹ 2,63,217
        </td>
        <td
          className="c3"
        >
          ₹ 5,783
        </td>
        <td
          className="c3"
        >
          ₹ 2,69,000
        </td>
        <td
          className="c3"
        >
          ₹ 0
        </td>
      </tr>
    </tbody>
  </table>
  <table
    className="c1"
  >
    <thead>
      <tr>
        <th
          className="c2"
        >
          Months
        </th>
        <th
          className="c2"
        >
          Principal
        </th>
        <th
          className="c2"
        >
          Interest
        </th>
        <th
          className="c2"
        >
          EMI
        </th>
        <th
          className="c2"
        >
          Balance
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          className="c3"
        >
          Month: 
          1
        </td>
        <td
          className="c3"
        >
          ₹ 58,500
        </td>
        <td
          className="c3"
        >
          ₹ 8,750
        </td>
        <td
          className="c3"
        >
          ₹ 67,250
        </td>
        <td
          className="c3"
        >
          ₹ 9,41,500
        </td>
      </tr>
      <tr>
        <td
          className="c3"
        >
          Month: 
          2
        </td>
        <td
          className="c3"
        >
          ₹ 59,012
        </td>
        <td
          className="c3"
        >
          ₹ 8,238
        </td>
        <td
          className="c3"
        >
          ₹ 67,250
        </td>
        <td
          className="c3"
        >
          ₹ 8,82,488
        </td>
      </tr>
      <tr>
        <td
          className="c3"
        >
          Month: 
          3
        </td>
        <td
          className="c3"
        >
          ₹ 59,528
        </td>
        <td
          className="c3"
        >
          ₹ 7,722
        </td>
        <td
          className="c3"
        >
          ₹ 67,250
        </td>
        <td
          className="c3"
        >
          ₹ 8,22,960
        </td>
      </tr>
      <tr>
        <td
          className="c3"
        >
          Month: 
          4
        </td>
        <td
          className="c3"
        >
          ₹ 60,049
        </td>
        <td
          className="c3"
        >
          ₹ 7,201
        </td>
        <td
          className="c3"
        >
          ₹ 67,250
        </td>
        <td
          className="c3"
        >
          ₹ 7,62,911
        </td>
      </tr>
      <tr>
        <td
          className="c3"
        >
          Month: 
          5
        </td>
        <td
          className="c3"
        >
          ₹ 60,575
        </td>
        <td
          className="c3"
        >
          ₹ 6,675
        </td>
        <td
          className="c3"
        >
          ₹ 67,250
        </td>
        <td
          className="c3"
        >
          ₹ 7,02,336
        </td>
      </tr>
      <tr>
        <td
          className="c3"
        >
          Month: 
          6
        </td>
        <td
          className="c3"
        >
          ₹ 61,105
        </td>
        <td
          className="c3"
        >
          ₹ 6,145
        </td>
        <td
          className="c3"
        >
          ₹ 67,250
        </td>
        <td
          className="c3"
        >
          ₹ 6,41,231
        </td>
      </tr>
      <tr>
        <td
          className="c3"
        >
          Month: 
          7
        </td>
        <td
          className="c3"
        >
          ₹ 61,639
        </td>
        <td
          className="c3"
        >
          ₹ 5,611
        </td>
        <td
          className="c3"
        >
          ₹ 67,250
        </td>
        <td
          className="c3"
        >
          ₹ 5,79,592
        </td>
      </tr>
      <tr>
        <td
          className="c3"
        >
          Month: 
          8
        </td>
        <td
          className="c3"
        >
          ₹ 62,179
        </td>
        <td
          className="c3"
        >
          ₹ 5,071
        </td>
        <td
          className="c3"
        >
          ₹ 67,250
        </td>
        <td
          className="c3"
        >
          ₹ 5,17,413
        </td>
      </tr>
      <tr>
        <td
          className="c3"
        >
          Month: 
          9
        </td>
        <td
          className="c3"
        >
          ₹ 62,723
        </td>
        <td
          className="c3"
        >
          ₹ 4,527
        </td>
        <td
          className="c3"
        >
          ₹ 67,250
        </td>
        <td
          className="c3"
        >
          ₹ 4,54,690
        </td>
      </tr>
      <tr>
        <td
          className="c3"
        >
          Month: 
          10
        </td>
        <td
          className="c3"
        >
          ₹ 63,271
        </td>
        <td
          className="c3"
        >
          ₹ 3,979
        </td>
        <td
          className="c3"
        >
          ₹ 67,250
        </td>
        <td
          className="c3"
        >
          ₹ 3,91,419
        </td>
      </tr>
      <tr>
        <td
          className="c3"
        >
          Month: 
          11
        </td>
        <td
          className="c3"
        >
          ₹ 63,825
        </td>
        <td
          className="c3"
        >
          ₹ 3,425
        </td>
        <td
          className="c3"
        >
          ₹ 67,250
        </td>
        <td
          className="c3"
        >
          ₹ 3,27,594
        </td>
      </tr>
      <tr>
        <td
          className="c3"
        >
          Month: 
          12
        </td>
        <td
          className="c3"
        >
          ₹ 64,384
        </td>
        <td
          className="c3"
        >
          ₹ 2,866
        </td>
        <td
          className="c3"
        >
          ₹ 67,250
        </td>
        <td
          className="c3"
        >
          ₹ 2,63,210
        </td>
      </tr>
      <tr>
        <td
          className="c3"
        >
          Month: 
          13
        </td>
        <td
          className="c3"
        >
          ₹ 64,947
        </td>
        <td
          className="c3"
        >
          ₹ 2,303
        </td>
        <td
          className="c3"
        >
          ₹ 67,250
        </td>
        <td
          className="c3"
        >
          ₹ 1,98,263
        </td>
      </tr>
      <tr>
        <td
          className="c3"
        >
          Month: 
          14
        </td>
        <td
          className="c3"
        >
          ₹ 65,515
        </td>
        <td
          className="c3"
        >
          ₹ 1,735
        </td>
        <td
          className="c3"
        >
          ₹ 67,250
        </td>
        <td
          className="c3"
        >
          ₹ 1,32,748
        </td>
      </tr>
      <tr>
        <td
          className="c3"
        >
          Month: 
          15
        </td>
        <td
          className="c3"
        >
          ₹ 66,088
        </td>
        <td
          className="c3"
        >
          ₹ 1,162
        </td>
        <td
          className="c3"
        >
          ₹ 67,250
        </td>
        <td
          className="c3"
        >
          ₹ 66,660
        </td>
      </tr>
      <tr>
        <td
          className="c3"
        >
          Month: 
          16
        </td>
        <td
          className="c3"
        >
          ₹ 66,667
        </td>
        <td
          className="c3"
        >
          ₹ 583
        </td>
        <td
          className="c3"
        >
          ₹ 67,250
        </td>
        <td
          className="c3"
        >
          ₹ 0
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
